src/wp-content/themes/twentyeleven/style.css

```diff
@@ -2487,7 +2479,7 @@ p.comment-form-comment {
 	.singular .entry-meta .edit-link a {
 		left: 0;
 		position: absolute;
-		top: 40px;
+		top: 20px;
```

Comment:
The Edit link placement would be closer to the current spot by changing the value to `0`, but I think 20 pixels is a bit safer with the possibility of custom CSS. Instead of subtracting 40 pixels at both breakpoints, I cut each in half.